apps/testingaccessibility/src/pages/api/auth/[...nextauth].ts

```diff
@@ -10,6 +10,53 @@ import {getSdk} from '../../../lib/prisma-api'
 import {defineRulesForPurchases} from '../../../server/ability'
 import mjml2html from 'mjml'
 
+export type MagicLinkEmailType =
+  | 'login'
+  | 'signup'
+  | 'reset'
+  | 'purchase'
+  | 'upgrade'
+
+export const sendVerificationRequest = async (params: {
+  identifier: string
+  url: string
+  expires: Date
+  provider: EmailConfig
+  token: string
+  type?: MagicLinkEmailType
+}) => {
+  const {
+    identifier: email,
+    url,
+    provider: {server, from},
+  } = params
+  const {host} = new URL(url)
+  const transport = createTransport(server)
+  const {getUserByEmail} = getSdk()
+
+  let subject
+
+  switch (params.type) {
+    case 'purchase':
+      subject = `Thank you for Purchasing Testing Accessibility (${host})`
+      break
+    default:
+      subject = `Log in to Testing Accessibility (${host})`
+  }
```

Comment:
I didn't change the template, but now we could use alternative html/text based on type which I think would be very nice.

Reply:
![Screen Shot 2022-07-06 at 15 27 59](https://user-images.githubusercontent.com/86834/177653888-616ec596-0bdf-40a5-a2b1-fdebccf3e603.png)